doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,193 @@
+"""
+============================================================
+Intravoxel incoherent motion
+============================================================
+The intravoxel incoherent motion (IVIM) model describes diffusion
+and perfusion in the signal acquired with diffusion. The IVIM model
+can be understood as an adaptation of Stejskals and Tanners work (ref)
+on biologic tissue, and was proposed by Le Bihan et al (ref).
+The hypothesis is that two compartments exist: a slow moving compartment,
+where particles diffuse in a Brownian fashion as a consequence of thermal
+energy, and a fast moving compartment (the vascular compartment), where
+blood moves as a consequence of pressure gradient. In this second
+compartment, a pseudo diffusion term (D* ) is introduced that describes
+on a macroscopic level the displacement of the blood elements in an
+assumed randomly laid vascular network. For the perfusion to have a
+physiologic meaning, one expects that D* is greater than D.
+
+The IVIM model expresses the diffusion signal as follows:
+
+ .. math: :
+
+    S(b) = S_{0}[fe ^ {-b*D_star} + (1 - f)e^{-b*D}]
+
+where $\mathbf{b}$ is the gradient value (which is dependent on
+the measurement parameters), $S_0$ is the signal in the absence
+of diffusion gradient sensitization, $\mathbf{f}$ is the perfusion
+fraction, $\mathbf{D}$ is the diffusion coefficient,
+ $\mathbf{D * }$ is the pseudo diffusion constant.
+
+In the following example we show how to fit the IVIM model
+on diffusion  weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import numpy as np
+from scipy.optimize import leastsq
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, _ivim_error, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using dipy's data fetcher read_ivim.
+The image data is an acquired IVIM dataset with 21 b - values
```

Comment:
=> "... This dataset was acquired with 21 b-values in 3 different directions. Volumes corresponding to different directions were registered to each other, and averaged across directions. Thus, this dataset has 4 dimensions, with the length of the last dimension corresponding to the number of b-values"